packages/ckeditor5-editor-multi-root/src/multirooteditoruiview.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module editor-multi-root/multirooteditoruiview
+ */
+
+import { EditorUIView, InlineEditableUIView, ToolbarView } from 'ckeditor5/src/ui';
+import type { Locale } from 'ckeditor5/src/utils';
+import type { View } from 'ckeditor5/src/engine';
+
+/**
+ * The multi-root editor UI view. It is a virtual view providing an inline
+ * {@link module:editor-multi-root/multirooteditoruiview~MultiRootEditorUIView#editable} and a
+ * {@link module:editor-multi-root/multirooteditoruiview~MultiRootEditorUIView#toolbar}, but without any
+ * specific arrangement of the components in the DOM.
+ *
+ * See {@link module:editor-multi-root/multirooteditor~MultiRootEditor.create `MultiRootEditor.create()`}
+ * to learn more about this view.
+ */
+export default class MultiRootEditorUIView extends EditorUIView {
+	/**
+	 * The main toolbar of the multi-root editor UI.
+	 */
+	public readonly toolbar: ToolbarView;
+
+	/**
+	 * Editable elements used by the multi-root editor UI.
+	 */
+	public readonly editables: Record<string, InlineEditableUIView>;
+
+	public readonly editable: InlineEditableUIView;
```

Comment:
This property doesn't make much sense for multi-root, where you have multiple editables, as above. This is set to the first defined editable.

I tried not extending `EditorUIView` but this got me into other typing problems (i.e. `EditorUI` requires `EditorUIView`, while `Editor` requires `EditorUI`, etc.).

Another idea would be to remove `editable` from `EditorUIView` and have it defined in particular `EditorUIView` implementations. This would require changes in places that uses `editable`. Alternatively, `editable` could be allowed to be `null`, then in places that requires it, we would have to tell TS that `editable` is set. AFAIR, `editable` is used in the balloon editor.

Of course, yet another idea would be to introduce hierarchy like I proposed for editor configs, so have separate ui view class for single and multi roots.

But I think that `editable` actually may not be needed at all. First of all, we have `getEditableElement()` on `EditorUI` that supports multiple editables (multi-roots). We could have `getEditable()` as well. On `EditorUI` or `EditorUIView`. We could also introduce `getFocusedEditable()` if there's a need to get the focused editable specifically (doesn't matter for single-editable editors, might matter for multi-root).

I do not think that we need to solve `EditorConfig` issues/changes; but this one is something that we might want to solve. `editor.ui.editable` does not make sense for multi-root, it smells. If you have proposals or strong opinion, I listen.

In general, I think there's quite a mess when it comes to which property is on what class and there are multiple ways to access some objects. E.g. we have `sourceElement`, we have `editor.ui.view.editable.element` we have `getEditableElement()` etc. I also don't understand why some things are on `EditorUI` and why other are on `EditorUIView`. However, this is not in a scope of this PR.

Reply:
Did you checked how `SourceEditing` works with the current solution? I can see the button is active (but should not) but nothing happens when clicked. This looks like related to `editor.ui.view.editable`.

Reply:
This PR is not supposed to make multi-root editor work with all the features, there will be follow-ups later.

I didn't check how source editing worked with earlier multi-roots but AFAICS in the code, it should work the same as earlier.